src/controller/tests/data_model/TestRead.cpp

```diff
@@ -109,6 +112,49 @@ const MockNodeConfig & TestMockNodeConfig()
     return config;
 }
 
+// Customizes attributes to pretend they are always OK to read
+class ReadDataModel : public CustomDataModel
+{
+public:
+    std::optional<DataModel::AttributeInfo> GetAttributeInfo(const ConcreteAttributePath & path) override
+    {
+        auto value = CustomDataModel::GetAttributeInfo(path);
+        if (value.has_value())
+        {
+            return value;
+        }
+
+        // TODO: figure out why these are special and fix up the test. This mock seems
+        //       nonsense, however adding actual paths into the mock config changes existing
+        //       test behaviour.
+        //
+        // These specific paths were obtained by grepping logs for what fake attributes we
+        // attempt to get info for within this test.
+        const ConcreteAttributePath kFakePaths[] = {
+            ConcreteAttributePath(kTestEndpointId, MockClusterId(2), 1),
+            ConcreteAttributePath(kTestEndpointId, MockClusterId(5), MockAttributeId(1)),
+            ConcreteAttributePath(kMockEndpoint1, MockClusterId(2), MockAttributeId(2)),
+            ConcreteAttributePath(kMockEndpoint3, MockClusterId(3), MockAttributeId(1)),
```

Comment:
Hold comment until I understand these more: it seems that some of these arise from path expansion (when we set endpoint to kInvalidEndpointId for example) and in that case it may hint that ACL error handling is not yet ok.

Reply:
Fixed up the logic: the Engine logic did not have sufficient information to determine unsupported attribute status (since it needs to know EP or Cluster or Attribute errors).